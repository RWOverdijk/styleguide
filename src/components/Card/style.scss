/** @define Card */

@import "../../css/variables";

.Van-Card {
  position: relative;
  padding: 25px 40px;
  background-color: $color-white;

  &--withShadow {
    box-shadow: 0 3px 9px rgba(0, 0, 0, 0.25);
  }

  @media (max-width: $container-sm) {
    padding-left: 15px;
    padding-right: 15px;
  }
}
