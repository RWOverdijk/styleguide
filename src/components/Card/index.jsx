import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class Card extends PureComponent {
  render() {
    const { className, shadow, children, ...props } = this.props;

    return (
      <div
        className={classnames('Van-Card', className, {
          'Van-Card--withShadow': shadow
        })}
        {...props}
      >
        {children}
      </div>
    );
  }
}

Card.defaultProps = {
  className: '',
  shadow: false,
  children: ''
};

Card.propTypes = {
  className: PropTypes.string,
  shadow: PropTypes.bool,
  children: PropTypes.node
};

export default Card;
