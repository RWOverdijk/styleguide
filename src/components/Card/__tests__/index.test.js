import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../';

describe("<Card>", () => {
  it("renders default Card", () => {
    const tree = renderer.create(
      <Card>My card</Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders Card with a shadow", () => {
    const tree = renderer.create(
      <Card shadow>My card</Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Card className="my-class">My card</Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Card data-id="0">My card</Card>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
