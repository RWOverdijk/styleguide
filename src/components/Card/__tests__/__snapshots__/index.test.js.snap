// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> renders Card with a shadow 1`] = `
<div
  className="Van-Card Van-Card--withShadow"
>
  My card
</div>
`;

exports[`<Card> renders default Card 1`] = `
<div
  className="Van-Card"
>
  My card
</div>
`;

exports[`<Card> renders with specific attributes 1`] = `
<div
  className="Van-Card"
  data-id="0"
>
  My card
</div>
`;

exports[`<Card> renders with specific className 1`] = `
<div
  className="Van-Card my-class"
>
  My card
</div>
`;
